File: python/Colleen.py
#!/usr/bin/env python3

"""
wow
"""

def colleen():
    """
    such piton
    """

    s = "#!/usr/bin/env python3{1}{1}{0}{0}{0}{1}wow{1}{0}{0}{0}{1}{1}def colleen():{1}    {0}{0}{0}{1}    such piton{1}    {0}{0}{0}{1}{1}    s = {0}{2}{0}{1}{1}    print(s.format(chr(34), chr(10), s)){1}{1}def main():{1}    colleen(){1}{1}if __name__ == {0}__main__{0}:{1}    main()"

    print(s.format(chr(34), chr(10), s))

def main():
    colleen()

if __name__ == "__main__":
    main()

File: perl/Colleen.pl
#!/usr/bin/env perl

use strict;
use warnings;

#
#	what the hell kinda language is this
#

sub colleen {
	#
	#	fancy some unicode? 🫠
	#
	my $s = '#!/usr/bin/env perl%1$c%1$cuse strict;%1$cuse warnings;%1$c%1$c#%1$c#%2$cwhat the hell kinda language is this%1$c#%1$c%1$csub colleen {%1$c%2$c#%1$c%2$c#%2$cfancy some unicode? 🫠%1$c%2$c#%1$c%2$cmy $s = %3$c%4$s%3$c;%1$c%1$c%2$cprintf "$s", 10, 9, 39, "$s";%1$c}%1$c%1$csub main {%1$c%2$ccolleen;%1$c}%1$c%1$cmain();%1$c';

	printf "$s", 10, 9, 39, "$s";
}

sub main {
	colleen;
}

main();
